main.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: false
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    x = 0
+    while x < self.length
+      yield(self[x])
+      x += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    x = 0
+    while x < self.length
+      yield(self[x], x)
+      x += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(method) if !block_given?
+
+    arr = []
+    x = 0
+    while x < self.length
+      if yield(self[x])
+        arr << self[x]
+      end
+      x += 1
+    end
+    arr
+  end
+
+  def my_all?(pattern = nil)
+    result = true
+    if block_given?
+      my_each { |ele| result &= (yield ele) }
+    elsif pattern
+      my_each { |ele| result &= pattern === ele }
+    else
+      my_each { |ele| result &= ele }
+    end
+    result
+  end
+
+  def my_any?(pattern = nil)
+    result = false
+    if block_given?
+      my_each { |ele| result = (yield ele) }
+    elsif pattern
+      my_each { |ele| result = pattern === ele }
+    else
+      my_each { |ele| result = ele }
+    end
+    result
+  end
+
+  def my_none?(pattern = nil)
+    result = true
+    if block_given?
+      my_each { |ele| result &= !(yield ele) }
+    elsif pattern
+      my_each { |ele| result &= pattern != ele }
+    else
+      my_each { |ele| result &= !ele }
+    end
+    result
+  end
```

Comment:
1. If a class is passed as argument, it should return true if none of collection is not part of class. but your return true for expected false case

2. If you pass a regex it should return true if nothing matches,but yours return true for expected false case